trustyai-python/trustyai-explainability
# FAI-902: Initial implementation of data-based fairness metrics

[FAI-902](https://issues.redhat.com/browse/FAI-902) Initial implementation of data-based fairness metrics

* Add SPD with Dataframes for pre-calculated groups
* Add group disparate impact ratio
* Add confusion matrix object
* Replace CopyOnWrite lists with ArrayList
* Add simple biased model to tests
* Add row-preserving test for Dataframes
* Add AOD test for Dataframes
* Add average predictive value difference
* Parallelise row access in Dataframes
* Move group fairness tests to proper package
* Add dataframe creation from inputs
* Add SPD for samples and models
* Add model-based SPD calling data-based SPD
* Add DIR dataframe for provided models
* Add model-based AOD
* Add model-based APVD

> Many thanks for submitting your Pull Request :heart:!
> 
> Please make sure that your PR meets the following requirements:
> 
> - [x] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
> - [x] Pull Request title is properly formatted: `FAI-XYZ Subject`
> - [x] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
> - [x] Pull Request contains link to the JIRA issue
> - [x] Pull Request contains link to any dependent or related Pull Request
> - [x] Pull Request contains description of the issue
> - [x] Pull Request does not include fixes for issues other than the main ticket
